### ISSUE TITLE
MySQL server has gone away after memory leak

### ISSUE THREAD

**Issue by ISekan** [Cacti/cacti#5261]
Centos7.9
5 CPU Intel(R) Xeon(R) CPU E7- 2870  @ 2.40GHz
6GB RAM
MariaDB 10.2.33
PHP Version  7.4.12
local poller
cacti update 1.2.14 -> 1.2.24 

After a while the poller shows errors as you see below, cacti.log grows very fast and swap usage is high. As result - lost data. I tried to adjust mysql setting, but it doesn't help.
If cacti.log is disabled, system works fine, but lost some data during boost-poller working

07-Mar-2023 02:45:11 - CMDPHP SQL Backtrace: (/graph_json.php[159]:rrdtool_function_graph(), /lib/rrd.php[1660]:variable_bandwidth_summation(), /lib/graph_variables.php[607]:bandwidth_summation(), /lib/graph_variables.php[333]:rrdtool_function_fetch(), /lib/rrd.php[942]:boost_fetch_cache_check(), /lib/boost.php[299]:boost_process_poller_output(), /lib/boost.php[773]:db_execute_prepared())
07-Mar-2023 02:45:11 - CMDPHP ERROR: A DB Exec Failed!, Error: Table 'cacti.poller_output_boost_arch_1678117309' doesn't exist
07-Mar-2023 02:45:32 - CMDPHP ERROR: A DB Row Failed!, Error: MySQL server has gone away
07-Mar-2023 02:45:32 - CMDPHP ERROR: A DB Row Failed!, Error: MySQL server has gone away
07-Mar-2023 02:45:32 - CMDPHP ERROR: A DB Row Failed!, Error: MySQL server has gone away
07-Mar-2023 02:45:32 - CMDPHP ERROR: A DB Row Failed!, Error: MySQL server has gone away
07-Mar-2023 02:45:32 - CMDPHP ERROR: A DB Row Failed!, Error: MySQL server has gone away
07-Mar-2023 02:45:32 - CMDPHP ERROR: A DB Row Failed!, Error: MySQL server has gone away

![image](https://user-images.githubusercontent.com/17400240/223324657-41876673-fa8d-4ae0-a18a-16e8e8ac0e24.png)
![image](https://user-images.githubusercontent.com/17400240/223325196-951d3566-6639-4586-b0b0-c08666e2120b.png)

**Comment by TheWitness** [comment]
You are running out of MySQL connections.  Spine design behavior changed in recent releases.  It now allocates a connection pool, which means if your have multiple collectors with dozens of threads each, the spine collector will open them all and use them throughout the data collection session and close them at the end.  This may leave not enough connections for other tasks.

There is a technical support page that should give you a bit more information on how close to the line you are, and the number can vary greatly depending on how many users you have accessing graphs concurrently.

![image](https://user-images.githubusercontent.com/1439914/223433863-fe3164c6-4a06-4116-a616-d642779c5460.png)

**Comment by ISekan** [comment]
Main Server: Current: 115, Min Required: 115
Overload happens during freeing up memory (see the graph) and log file with a million lines "CMDPHP ERROR: A DB Row Failed!, Error: MySQL server has gone away" growed almost 1GB for half day!
I increased connections to 400. Will see. Main Server: Current: 400, Min Required: 115
Thanks, TheWitness!!
![image](https://user-images.githubusercontent.com/17400240/223619628-47f053fb-f862-43f1-9002-b3c0636090b1.png)

**Comment by ISekan** [comment]
Increase connection doesn't help
I have only one local poller
![image](https://user-images.githubusercontent.com/17400240/223669809-1716fe8d-7534-41ab-a697-f8929849a812.png)

**Comment by bmfmancini** [comment]
paste your mysql config here

**Comment by ISekan** [comment]
> paste your mysql config here

[mariadb]
innodb_io_capacity = 5000
innodb_io_capacity_max = 10000
max_connections = 400
max_heap_table_size = 92M
tmp_table_size = 92M
max_allowed_packet = 16M
innodb_buffer_pool_size = 1024M
innodb_buffer_pool_instances = 10
innodb_doublewrite = off
innodb_flush_log_at_timeout = 4
innodb_read_io_threads = 34
innodb_write_io_threads = 17
innodb_log_file_size = 128M
innodb_flush_log_at_trx_commit = 2
join_buffer_size = 2M
sort_buffer_size = 2M
innodb_large_prefix = 1
wait_timeout = 180
connect_timeout = 180
sql_mode=NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION
query_cache_size = 0
query_cache_type = 0
innodb_lock_wait_timeout = 60
table_cache = 200

p.s. I switched off the boost, and the same situation..

**Comment by bmfmancini** [comment]
Are you using cactid?

**Comment by TheWitness** [comment]
Also, a 2002 error likely means that the database is crashing and restarting.  You watch the MySQL/MariaDB log file for crashes.

**Comment by TheWitness** [comment]
Upgrading to 10.11.2+ worked real well for me.  I use 10.5.13 in production.  Very large 35TB Cacti database.

**Comment by TheWitness** [comment]
MariaDB provides a nice yum configuration script.

**Comment by ISekan** [comment]
The problem has been solved with reducing setting **innodb_buffer_pool_size from 1024M to 128 M** 
Although, according the recommendation of cacti, it has to been set 1.5GB  (my config: spine 1min, 5CPU, 6GB RAM, /var/lib/mysql/cacti - 550MB)

p.s. I updated mariaDB to 10.11.2 on another one server.  InnoDB enable by default. What it means and what I should add to my.cnf.d?
![image](https://user-images.githubusercontent.com/17400240/224660557-13abfbe8-7991-409d-b70c-875e53d32440.png)
